ui/app/models/secret-engine.js

```diff
@@ -22,127 +22,194 @@ const validations = {
       'The KV Secrets Engine can operate in different modes. Version 1 is the original generic Secrets Engine the allows for storing of static key/value pairs. Version 2 added more features including data versioning, TTLs, and check and set.',
     possibleValues: [2, 1],
     // This shouldn't be defaultValue because if no version comes back from API we should assume it's v1
     defaultFormValue: 2, // Set the form to 2 by default
-  }),
-  config: belongsTo('mount-config', { async: false, inverse: null }),
-  local: attr('boolean', {
-    helpText:
-      'When Replication is enabled, a local mount will not be replicated across clusters. This can only be specified at mount time.',
-  }),
-  sealWrap: attr('boolean', {
-    helpText:
-      'When enabled - if a seal supporting seal wrapping is specified in the configuration, all critical security parameters (CSPs) in this backend will be seal wrapped. (For K/V mounts, all values will be seal wrapped.) This can only be specified at mount time.',
-  }),
+  })
+  version;
+
+  // SSH specific attributes
+  @attr('string') privateKey;
+  @attr('string') publicKey;
+  @attr('boolean', {
+    defaultValue: true,
+  })
+  generateSigningKey;
+
+  // AWS specific attributes
+  @attr('string') lease;
+  @attr('string') leaseMax;
+
+  // Returned from API response
+  @attr('string') accessor;
+
   // KV 2 additional config default options
-  maxVersions: attr('number', {
+  @attr('number', {
     defaultValue: 0,
     label: 'Maximum number of versions',
     subText:
       'The number of versions to keep per key. Once the number of keys exceeds the maximum number set here, the oldest version will be permanently deleted. This value applies to all keys, but a key’s metadata settings can overwrite this value. When 0 is used or the value is unset, Vault will keep 10 versions.',
-  }),
-  casRequired: attr('boolean', {
+  })
+  maxVersions;
+  @attr('boolean', {
     defaultValue: false,
     label: 'Require Check and Set',
     subText:
       'If checked, all keys will require the cas parameter to be set on all write requests. A key’s metadata settings can overwrite this value.',
-  }),
-  deleteVersionAfter: attr({
+  })
+  casRequired;
+  @attr({
     defaultValue: 0,
     editType: 'ttl',
     label: 'Automate secret deletion',
     helperTextDisabled: 'A secret’s version must be manually deleted.',
     helperTextEnabled: 'Delete all new versions of this secret after',
-  }),
+  })
+  deleteVersionAfter;
+
+  /* GETTERS */
+  get modelTypeForKV() {
+    const engineType = this.engineType;
+    if ((engineType === 'kv' || engineType === 'generic') && this.version === 2) {
+      return 'secret-v2';
+    }
+    return 'secret';
+  }
+  get isV2KV() {
+    return this.modelTypeForKV === 'secret-v2';
+  }
+
+  get attrs() {
+    return this.formFields.map((fieldName) => {
+      return this.allByKey[fieldName];
+    });
+  }
+
+  get fieldGroups() {
+    return this.formFieldGroups.map((group) => {
```

Comment:
This is not very pretty to look at -- could maybe add a helper function within the decorator for managing fieldGroup objects => expanded fieldGroups? Might be useful if we end up using this decorator pattern more.

Reply:
I like that idea. Moving the logic to a decorator also allows us to give context as you have in the decorator you created.

Reply:
Done!